plugins/inputs/prometheus/prometheus.go

```diff
@@ -351,7 +352,15 @@ func (p *Prometheus) Gather(acc telegraf.Accumulator) error {
 		wg.Add(1)
 		go func(serviceURL URLAndAddress) {
 			defer wg.Done()
-			acc.AddError(p.gatherURL(serviceURL, acc))
+			requestFields, tags, err := p.gatherURL(serviceURL, acc)
+			if err != nil {
+				acc.AddError(err)
+			}
```

Comment:
You can simply do
```suggestion
			acc.AddError(err)
```
because this function internally checks for the error to be non-nil.